Library/Homebrew/requirements/macos_requirement.rb

```diff
@@ -22,6 +22,10 @@ def initialize(tags = [], comparator: ">=")
 
     @comparator = comparator
     super(tags)
```

Comment:
Might be best to wrap it so it doesn't skip `super`.

Reply:
@Bo98 Looking at this: is an instance of this class even going to be usable at all if `@version` is `nil`?

Reply:
Without a `@version` it should behave like `depends :macos`. Which thinking about it is not correct on Linux.

As long as it properly no-ops it's fine. Deprecation shouldn't mean the formula is unusable.

Reply:
> so this might need extra logic.

We didn't do that last time and didn't have any complaints so I'm really not sure it's worth the extra effort here 🤷🏻

Reply:
Yeah probably because we remove them from all our taps prior to deprecation.

Reply:
I've pushed an amendment - feel free to re-push your local version if you disagree.

It's only really correct if `comparator` is `>=`, but nothing was right before anyway for `==` or `<=` so that hasn't really regressed.